### PR TITLE
incremental(windows) add TransmitFile and some fixes for fetch, h2 and http

### PR DESCRIPTION
**Pull request by cirospaciari**
### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

Existing tests

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1297,6 +1297,7 @@ else()
         ucrt
         userenv
         dbghelp
+        wsock32 # ws2_32 required by TransmitFile aka sendfile on windows
```
Comment (Jarred-Sumner) [inline CMakeLists.txt:1314]: I wish we could lazily load these

**Comment by Jarred-Sumner** [comment]
windows has build errors

**Comment by cirospaciari** [comment]
> windows has build errors



> windows has build errors

Yeah main was not building on windows so I open https://github.com/oven-sh/bun/pull/8100

File: src/http.zig [hunk 2]
```diff
@@ -115,6 +115,16 @@ pub const HTTPRequestBody = union(enum) {
     }
 };
 
+pub fn canUseBrotli() bool {
```
Comment (Jarred-Sumner) [inline src/http.zig:118]: this is no longer used

**Comment by Jarred-Sumner** [comment]
thank you

**Pull request merged**